### PR TITLE
Add model.

### PR DESCRIPTION
**Pull request by ashdnazg**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+mod model;
```
Comment (nihohit) [inline src/main.rs:2]: doesn't `mod` open a module? so shouldn't it be in model.rs?

https://doc.rust-lang.org/std/keyword.mod.html

File: src/model.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+pub enum Advice {
+	GetToCornerToWin,
+	DoSomething,
+	DoSomethingElse
+}
+
+pub struct Tile {
+	color: TileColor,
+	advice: Option<Advice>,
```
Comment (nihohit) [inline src/model.rs]: why is this optional? also, why isn't this a string? assuming that each tile will have a unique message, is an enum worth the effort?

File: src/model.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	passable: bool
+}
+
+pub struct Board {
+	tiles: Vec<Option<Tile>>,
```
Comment (nihohit) [inline src/model.rs]: maybe multidimensional array?

**Review by nihohit — APPROVED** [review]

File: src/model.rs (hunk 2 above)
Comment (ashdnazg) [reply src/model.rs]: I imagined there might be tiles with no advice, but I guess YAGNI.
I'll change it to non optional until we need it.

File: src/model.rs (hunk 3 above)
Comment (ashdnazg) [reply src/model.rs]: Makes sense.

**Pull request merged**